src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  AfterContentInit,
+  Attribute,
+  Component,
+  ContentChildren,
+  Directive,
+  HostBinding,
+  HostListener,
+  Input,
+  Optional,
+  Output,
+  QueryList,
+  ViewEncapsulation,
+  forwardRef
+} from 'angular2/core';
+
+import {EventEmitter} from 'angular2/src/facade/async';
+import {Event} from 'angular2/src/facade/browser';
+
+import {MdRadioDispatcher} from './radio_dispatcher';
+
+// TODO(mtlin):
+// Ink ripple is currently placeholder.
+// Determine motion spec for button transitions.
+// Design review.
+// RTL
+// Support forms API.
+// Use ChangeDetectionStrategy.OnPush
+
+var _uniqueIdCounter = 0;
+
+/** A simple change event emitted by either MdRadioButton or MdRadioGroup. */
+export class MdRadioChange {
+  source: MdRadioButton;
+  value: any;
+}
+
+@Directive({
+  selector: 'md-radio-group',
+  host: {
+    'role': 'radiogroup',
+  },
+})
+export class MdRadioGroup implements AfterContentInit {
+  /** The value for the radio group. Should match currently selected button. */
+  private _value: any = null;
+
+  /** The HTML name attribute applied to radio buttons in this group. */
+  private _name: string = null;
+
+  /** Disables all individual radio buttons assigned to this group. */
+  private _disabled: boolean = false;
+
+  /** The currently selected radio button. Should match value. */
+  private _selected: MdRadioButton = null;
+
+  /** Event emitted when the group value changes. */
+  @Output()
+  change: EventEmitter<MdRadioChange> = new EventEmitter();
+
+  /** Child radio buttons. */
+  @ContentChildren(forwardRef(() => MdRadioButton))
+  private _radios: QueryList<MdRadioButton> = null;
+
+  /**
+   * Initialize properties once content children are available.
+   * This allows us to propagate relevant attributes to associated buttons.
+   */
+  ngAfterContentInit() {
+    if (this._name == null) {
+      this.name = `md-radio-group-${_uniqueIdCounter++}`;
+    } else {
+      this._updateChildRadioNames();
+    }
+
+    if (this._disabled) {
+      this.disabled = true;
+    }
+  }
+
+  @Input()
+  get name(): string {
+    return this._name;
+  }
+
+  set name(value: string) {
+    this._name = value;
+
+    this._updateChildRadioNames();
+  }
+
+  /** Propagate name attribute to radio buttons. */
+  private _updateChildRadioNames() {
+    if (this._radios != null) {
+      this._radios.forEach((radio: MdRadioButton) => {
```

Comment:
TypeScript _should_ know the type of `radio` implicitly since `QueryList` is generic; does it work if you just do
```ts
this._radios.forEach(radio => {
  radio.name = this._name;
});
```
?